pycaret/internal/pipeline.py

```diff
@@ -79,6 +80,42 @@ def __init__(self, steps, *, memory=None, verbose=False):
         self._fit_vars = set()
         self._feature_names_in = None
 
+    def __getstate__(self):
+        try:
```

Comment:
I don't understand this if/else block.

When would `super().__getstate__()` raise an `AttributeError`? 

Also, by default, this seems to only save the state from the parent `imblearn.pipeline.Pipeline` and discards the state from the actual `Pipeline`?

Reply:
The `__getstate__()` is an optional special method, so super class does not necessarily have it. ( However, the current superclass, imblearn.pipeline.Pipeline, has it. )

The `self` in superclass's methods has attributes defined in its subclasses.
Thus, when the superclass executes `state = self.__dict__.copy()`, all attributes are saved.
However, since there is no guarantee that `state = self.__dict__.copy()` will be executed, I  commited some changes.